evaluation.md

```diff
@@ -6,14 +6,136 @@ _Describe each change from your ideal syntax to the syntax you implemented, and
+
+* We also realized why the original Picobot language included 2 integers to represent states: one to represent the current staet, and the other to represent the state that the machine should go to after the current command is executed.
+
+* In the case clas Free, which takes 4 strings and has a method called toFreeArray, we realized that a "Free" array can have 0, 1, 2, 3, of 4 parameters, representing how many States are free. We did not want to create 5 different case class objects representing the 5 possible constructors for the case class Free, so we changed the syntax (see Empty.scala) so that Free needed to take 4 String parameters. Below is an example of a command:
+
+> Condition(CurrentState(0), Free("W", "", "", ""), Occupied("", "", "", ""), Next("W"), 0).set.
+
+Whereas our original language would have been:
+ > Condition(CurrentState(0), Free("W"), Occupied(), Next("W"), 0).set, 
+ 
+ which allows both Free and Occupied to only anywhere between 0 and 4 parameters, inclusive. 
+
 
 **On a scale of 1–10 (where 10 is "very difficult"), how difficult was it to map your syntax to the provided API?**
 
+It took quite a bit of thinking (especially when it came to using case classes, abstract classes, defs, etc.) , but this is how we ended up mapping our syntax to the provided API:
+
+Within our InternalDSL class, we have a method called setSurroundings which takes two string arrays and returns a Surroundings object. The goal of this method is to take 2 string arrays, one representing the letters that are exclusively free (so not either free OR occupied) and the letters that are exclusively occupied. Our method achieves this by first creating an array of 4 RelativeDescriptions, the 0th index corresponding to the RelativeDescription of N that the user designates (either Anything, Occupied, or Free), and the 3rd index representing the RelativeDescription of S that the user designates (either Anything, Occupied, or Free). Then, we have 8 if statements to check whether N, E, W, or S is in the "Free" string array, and whether N, E, W, or S is in the "Occupied" string array. If, for example, W is in the "Free" string array, we would assign array[2] = Free to the array of 4 RelativeDescriptions. At the end, we return the array of 4 RelativeDescriptions, which is either populated with all "Anything"'s, or a combination of "Anything", "Free", and "Occupied". 
+
+We tried to make our language fit within the Scala syntax, but we didn't see pattern matching as an option, which, after looking at Prof. Ben's solutions, seems to have been one of the only ways to make our language work as an internal DSL. However, we ended up with an internal DSL that served as an intermediate DSL to our external DSL. Implementing the internal DSL was difficult -- we would give it an 8.
```

Comment:
It's nice that your internal DSL serves as an intermediate DSL your external DSL. Just looking at your internal DSL code, however, I would have never guessed it was as hard to do. That's easy for me to say though, since I'm already looking at the finished product